content/docs/testing-recipes.md

```diff
@@ -95,46 +78,42 @@ export default function User(props) {
 We can write tests for it:
 
 ```jsx
+// user.test.js
+import { render } from "@testing-library/react";
 import User from "./user";
-import { render } from "react-dom";
-import { act } from "react-dom/test-utils";
 
-let container = null
 beforeEach(() => {
-  // setup a DOM element as a render target
-  container = document.createElement('div')
-  document.body.appendChild(container)
-})
+  // mock window.fetch so the unit tests don't actually make HTTP requests
+  jest.spyOn(window, "fetch").mockImplementation(() => {});
+});
 
 afterEach(() => {
-  // cleanup on exiting
-  container.remove();
-  container = null  
-})
+  // remove the mock to ensure tests are completely isolated
+  window.fetch.mockRestore();
+});
 
-it("renders user data", async () => {
+test("renders user data", async () => {
   const fakeUser = {
     name: "Joni Baez",
     age: "32",
-    address: "123, Charming Avenue"
+    address: "123, Charming Avenue",
   };
-  jest.spyOn(global, "fetch").mockImplementation(() =>
+  window.fetch.mockImplementationOnce(() =>
     Promise.resolve({
-      json: () => Promise.resolve(fakeUser)
-    })
+      json: () => Promise.resolve(fakeUser),
+    }),
   );
 
-  // Use the asynchronous version of act to flush resolved promises
-  await act(async () => {
-    render(<User id="123" />, div);
-  });
+  const { getByLabelText, getByText } = render(<User id="123" />);
 
-  expect(container.querySelector("summary").textContent).toBe(fakeUser.name);
-  expect(container.querySelector("strong").textContent).toBe(fakeUser.age);
-  expect(container.textContent).toContain(fakeUser.address);
+  expect(window.fetch).toHaveBeenCalledTimes(1);
+  expect(window.fetch).toHaveBeenCalledWith("/123");
 
-  // remove the mock to ensure tests are completely isolated
-  global.fetch.mockRestore();
+  await waitForElementToBeRemoved(() => getByText(/loading/i));
```

Comment:
This is why we don't need `async act` 💯

Reply:
These async utilities have made a huge difference in my tests. Instead of manually "advancing" state, you await the condition. Also, the pattern isn't just for DOM changes, you can use `wait(retryThis)` for any expectation. 

```jsx
fireEvent.click(getByText(/submit/i))

// Don't care if this call happens
// synchronously or not
await wait(() => {
   expect(api.fetchData)
    .toHaveBeenCalledWith(id)
})

// the result of the call is new data
// appearing
await findByLabelText(/my data/i)
```

Reply:
Agreed. The async utilities are just as game changing as the queries IMO.